pkg/collect/host_certificate.go

```diff
@@ -0,0 +1,59 @@
+package collect
+
+import (
+	"bytes"
+	"crypto/tls"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+)
+
+const KeyPairMissing = "key-pair-missing"
+const KeyPairSwitched = "key-pair-switched"
+const KeyPairEncrypted = "key-pair-encrypted"
+const KeyPairMismatch = "key-pair-mismatch"
+const KeyPairInvalid = "key-pair-invalid"
+const KeyPairValid = "key-pair-valid"
+
+func HostCertificate(c *HostCollector) (map[string][]byte, error) {
+	var result = KeyPairValid
+
+	_, err := tls.LoadX509KeyPair(c.Collect.Certificate.CertificatePath, c.Collect.Certificate.KeyPath)
+	if err != nil {
+		if strings.Contains(err.Error(), "no such file") {
+			result = KeyPairMissing
+		} else if strings.Contains(err.Error(), "PEM inputs may have been switched") {
+			result = KeyPairSwitched
+		} else if strings.Contains(err.Error(), "found a certificate rather than a key") {
+			result = KeyPairSwitched
+		} else if strings.Contains(err.Error(), "private key does not match public key") {
+			result = KeyPairMismatch
+		} else if strings.Contains(err.Error(), "failed to parse private key") {
```

Comment:
I would greatly prefer to use `errors.Is` here for each of these `strings.Contains`

Reply:
I agree but I don't think it's possible since the standard library is using errors.New(string) https://golang.org/src/crypto/tls/tls.go?s=8380:8451#L299